src/process_data.c

```diff
@@ -47,13 +47,30 @@
 /*----------------------------------------------------------------------------*/
 /*                             Function Prototypes                            */
 /*----------------------------------------------------------------------------*/
-void __msgpack_pack_string( msgpack_packer *pk, const void *string, size_t n );
+/* None */
 
 /*----------------------------------------------------------------------------*/
 /*                             External Functions                             */
 /*----------------------------------------------------------------------------*/
 /* See process_data.h for details. */
-ssize_t process_message_cu( const char *filename, const char *md5, wrp_msg_t *cu )
+ssize_t process_create( const char *filename, const char *md5, wrp_msg_t *cu )
+{
+    ssize_t write_size = 0;
+    FILE *file_handle = NULL;
+
+    file_handle = fopen(filename, "r");
```

Comment:
Use `access()` instead if possible.  It also appears there is a file handle leak as `file_handle` is not closed if `fopen()` is successful.